label_studio/tests/ml/test_predict.py

```diff
@@ -0,0 +1,151 @@
+import json
+
+import pytest
+
+from label_studio.tests.utils import make_project, make_task, register_ml_backend_mock
+
+
+@pytest.fixture
+def ml_backend_for_test_predict(ml_backend):
+    # ML backend with single prediction per task
+    register_ml_backend_mock(
+        ml_backend,
+        url='http://localhost:9092',
+        predictions={
+            'results': [
+                {
+                    'model_version': 'ModelSingle',
+                    'score': 0.1,
+                    'result': [
+                        {'from_name': 'label', 'to_name': 'text', 'type': 'choices', 'value': {'choices': ['Single']}}
+                    ],
+                },
+            ]
+        },
+    )
+    # ML backend with multiple predictions per task
+    register_ml_backend_mock(
+        ml_backend,
+        url='http://localhost:9093',
+        predictions={
+            'results': [
+                [
+                    {
+                        'model_version': 'ModelA',
+                        'score': 0.2,
+                        'result': [
+                            {
+                                'from_name': 'label',
+                                'to_name': 'text',
+                                'type': 'choices',
+                                'value': {'choices': ['label_A']},
+                            }
+                        ],
+                    },
+                    {
+                        'model_version': 'ModelB',
+                        'score': 0.3,
+                        'result': [
+                            {
+                                'from_name': 'label',
+                                'to_name': 'text',
+                                'type': 'choices',
+                                'value': {'choices': ['label_B']},
+                            }
+                        ],
+                    },
+                ]
+            ]
+        },
+    )
+    yield ml_backend
+
+
+@pytest.mark.django_db
+def test_get_single_prediction_on_task(business_client, ml_backend_for_test_predict):
+    project = make_project(
+        config=dict(
+            is_published=True,
+            label_config="""
+                <View>
+                  <Text name="text" value="$text"></Text>
+                  <Choices name="label" choice="single">
+                    <Choice value="label_A"></Choice>
+                    <Choice value="label_B"></Choice>
+                  </Choices>
+                </View>""",
+            title='test_get_single_prediction_on_task',
+        ),
+        user=business_client.user,
+        use_ml_backend=False,
+    )
+
+    make_task({'data': {'text': 'test 1'}}, project)
+    make_task({'data': {'text': 'test 2'}}, project)
+    make_task({'data': {'text': 'test 3'}}, project)
```

Comment:
Why are three tasks created in this + the other testcase? It's not obvious to me how the downstream test code depends on this happening.

Reply:
fixed